src/OrchardCore.Modules/OrchardCore.Setup/Views/_Layout.cshtml

```diff
@@ -48,12 +49,7 @@
     <script>
         function setLocalizationUrl() {
             var culturesList = document.getElementById('culturesList');
-            var culture = culturesList.options[culturesList.selectedIndex].value
-            var queryString = window.location.search;
-            var searchParams = new URLSearchParams(queryString);
-
-            searchParams.set("ui-culture", culture);
-            window.location = culturesList.dataset.url + '?' + searchParams.toString();
+            window.location = culturesList.options[culturesList.selectedIndex].dataset.url;
```

Comment:
I thought that window loction would need to be set with an absolute uri, not a relative url

But didn't work on javascript for a while ;)

Reply:
Okay just tried, it works with the relative url, the only problem, as said above, is that we lose the token parameter of the current query string.

Reply:
I'm not sure what's the wrong with `URLSearchParams` that I did earlier?!!